src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -2222,7 +2222,7 @@ public static ExpressionSyntax GetStandaloneExpression(ExpressionSyntax expressi
         /// <summary>
         /// Given a conditional binding expression, find corresponding conditional access node.
         /// </summary>
-        internal static ConditionalAccessExpressionSyntax? FindConditionalAccessNodeForBinding(CSharpSyntaxNode node)
+        internal static ConditionalAccessExpressionSyntax FindConditionalAccessNodeForBinding(CSharpSyntaxNode node)
```

Comment:
>FindConditionalAccessNodeForBinding [](start = 58, length = 35)

I am not sure if we even need to make changes in this function. It feels like the problem is bigger. We shouldn't even attempt to speculate on a syntax node that cannot be bound on its own. For example, even if there is a conditional access expression that we can find, but SemanticModel was given only its child node to speculate on, SemanticModel should refuse to do so right away. So, the problem isn't just with null parent.

Reply:
It feels like a change should be made in CanGetSemanticInfo function, it already detects similar situations with object/collection initializers, etc. 

---
In reply to: [490571483](https://github.com/dotnet/roslyn/pull/46565#discussion_r490571483) [](ancestors = 490571483)